segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -531,9 +549,18 @@ private boolean handleClosedLedgers(List<Write> writes) {
             }
 
             // Write likely failed because of LedgerClosedException. Need to check the LastAddConfirmed for each
-            // involved Ledger and see if the write actually made it through or not.
+            // involved Ledger and see if the Write actually made it through or not.
             long lac = fetchLastAddConfirmed(w.getWriteLedger(), lastAddsConfirmed);
-            if (w.getEntryId() >= 0 && w.getEntryId() <= lac) {
+            // Note that there is a small chance that the Write has "partially" made it, which means that there is at
+            // least one Bookie with the entry stored, but the ackQuorum has not been satisfied. This implies that the
+            // addCallback() has not been called (i.e. entryId=Long.MIN_VALUE). For this particular case, we compare the
+            // expected entry id of the Write against LAC as a safety mechanism to prevent duplicate entries (see
+            // https://github.com/pravega/pravega/issues/6444).
+            boolean isWriteStoredInPartialAckQuorum = isWriteStoredInPartialAckQuorum(w, lac);
+            if ((w.getEntryId() >= 0 && w.getEntryId() <= lac) || isWriteStoredInPartialAckQuorum) {
```

Comment:
since its a partially acke'd entry, can it happen that while reading we read it from a BK replica which doesn't have this entry?(which would imply data loss)

Reply:
If the entry is at least in one Bookie, it will be read (the client requests the data to the whole ensemble and the first reply is the one that is served to the client).